src/BaseWebsiteCreator.ts

```diff
@@ -0,0 +1,690 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { AzureAccountWrapper } from './azureAccountWrapper';
+import { WizardBase, WizardResult, WizardStep, SubscriptionStepBase, QuickPickItemWithData } from './wizard';
+import { SubscriptionModels, ResourceManagementClient, ResourceModels } from 'azure-arm-resource';
+import WebSiteManagementClient = require('azure-arm-website');
+import * as WebSiteModels from '../node_modules/azure-arm-website/lib/models';
+import * as util from './util';
+
+export type WebsiteOS = "linux" | "windows";
+export type AppKind = "app" | "functionapp";
+
+export abstract class WebsiteCreatorBase extends WizardBase {
```

Comment:
Should the file name match the class name? (BaseWebsiteCreator vs WebsiteCreatorBase)